src/app/hooks/useRbfTransactionData.ts

```diff
@@ -0,0 +1,274 @@
+  const baseUrl = network?.address;
+  const apiUrl = `${baseUrl}/extended/v1/address/${stxAddress}/nonces`;
+  return axios.get<LatestNonceResponse>(apiUrl).then((response) => response.data);
+}
+
+interface RawTransactionResponse {
+  raw_tx: string;
+}
+
+export async function getRawTransaction(txId: string, network: SettingsNetwork): Promise<string> {
+  const baseUrl = network?.address;
+  const apiUrl = `${baseUrl}/extended/v1/tx/${txId}/raw`;
+
+  return axios.get<RawTransactionResponse>(apiUrl).then((response) => response.data.raw_tx);
+}
+
+const constructRecommendedFees = (
+  lowerName: keyof RbfRecommendedFees,
+  lowerFeeRate: number,
+  higherName: keyof RbfRecommendedFees,
+  higherFeeRate: number,
+  stxAvailableBalance: string,
+  thresholdHighStacksFee?: number,
+): RbfRecommendedFees => {
+  let lowerFee = lowerFeeRate;
+  let higherFee = higherFeeRate;
+
+  if (thresholdHighStacksFee) {
+    // adding a fee cap
+
+    if (higherFee > thresholdHighStacksFee) {
+      higherFee = thresholdHighStacksFee;
+    }
+
+    if (lowerFee > thresholdHighStacksFee) {
+      lowerFee = thresholdHighStacksFee * 0.75;
+    }
+  }
+
+  return {
+    [lowerName]: {
+      enoughFunds: BigNumber(lowerFee).lte(BigNumber(stxAvailableBalance)),
+      feeRate: microStxToStx(lowerFee).toNumber(),
+      fee: microStxToStx(lowerFee).toNumber(),
+    },
+    [higherName]: {
+      enoughFunds: BigNumber(higherFee).lte(BigNumber(stxAvailableBalance)),
+      feeRate: microStxToStx(higherFee).toNumber(),
+      fee: microStxToStx(higherFee).toNumber(),
+    },
+  };
+};
+
+const sortFees = (fees: RbfRecommendedFees) =>
+  Object.fromEntries(
+    Object.entries(fees).sort((a, b) => {
+      const priorityOrder = ['highest', 'higher', 'high', 'medium'];
+      return priorityOrder.indexOf(a[0]) - priorityOrder.indexOf(b[0]);
+    }),
+  );
+
+const useRbfTransactionData = (transaction?: BtcTransactionData | StxTransactionData): RbfData => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [rbfData, setRbfData] = useState<RbfData>({});
+  const { accountType, network, selectedAccount, stxAvailableBalance, feeMultipliers } =
+    useWalletSelector();
+  const seedVault = useSeedVault();
+  const btcClient = useBtcClient();
+  const selectedNetwork = useNetworkSelector();
+
+  // TODO: move the STX RBF calculations to xverse-core and add unit tests
+  const fetchStxData = useCallback(async () => {
+    if (!transaction || isBtcTransaction(transaction)) {
+      return;
+    }
+
+    try {
+      setIsLoading(true);
+
+      const fee = microStxToStx(transaction.fee);
+      const txRaw: string = await getRawTransaction(transaction.txid, network);
+      const unsignedTx: StacksTransaction = deserializeTransaction(txRaw);
+
+      const [slow, medium, high] = await estimateTransaction(
+        unsignedTx.payload,
+        undefined,
+        selectedNetwork,
+      );
+
+      const currentMicrostacksFee = stxToMicrostacks(fee);
```

Comment:
```suggestion
      const currentMicrostacksFee = transaction.fee;
```

yeah the minimum fee rounding is a lot cleaner, thanks!

nit: you still have this extra arithmetic